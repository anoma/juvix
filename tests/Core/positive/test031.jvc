-- Ackermann function

def ack := \m \n
  if m = 0 then
    n + 1
  else if n = 0 then
    ack (m - 1) 1
  else
    ack (m - 1) (ack m (n - 1));

def writeLn := \x write x >> write "\n";

writeLn (ack 0 7) >>
writeLn (ack 1 7) >>
writeLn (ack 1 13) >>
writeLn (ack 2 7) >>
writeLn (ack 2 13) >>
writeLn (ack 3 4)
