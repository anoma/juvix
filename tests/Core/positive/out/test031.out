8
9
15
17
29
1021
